File: src/main/java/com/larryhsiao/clotho/dgist/HexString.java
package com.larryhsiao.clotho.dgist;

import com.larryhsiao.clotho.Source;

import java.util.Formatter;

/**
 * Convert to hex from given bytes.
 */
public class HexString implements Source<String> {
    private final byte[] input;

    public HexString(byte[] input) {this.input = input;}

    @Override
    public String value() {
        final Formatter formatter = new Formatter();
        for (byte b : input) {
            formatter.format("%02x", b);
        }
        return formatter.toString();
    }
}

File: src/main/java/com/larryhsiao/clotho/dgist/HexString.kt
package com.larryhsiao.clotho.dgist

import com.larryhsiao.clotho.Source
import java.util.*

/**
 * Convert to hex from given data.
 */
class HexString(private val bytes: ByteArray) : Source<String> {
    override fun value(): String {
        with(Formatter()) {
            for (byte in bytes) {
                format("%02x", byte)
            }

            return toString()
        }
    }
}